__tests__/app.test.js

```diff
@@ -252,50 +252,97 @@ describe("/api/articles/:article_id", () => {
+    return request(app)
+      .delete("/api/comments/1000")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Comment does not exist");
+      });
+  });
+  test("DELETE: 400 sends an appropriate error status alongside an error message when given an invalid id", () => {
+    return request(app)
+      .delete("/api/comments/invalid-id")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
+});
+describe("/api/users", () => {
+  test("GET: 200 Sends an array of user objects with the properties of username, name and avatar", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.users.length).toBe(4);
+        body.users.forEach((user) => {
+          expect(typeof user.username).toBe("string");
+          expect(typeof user.name).toBe("string");
+          expect(typeof user.avatar_url).toBe("string");
+        });
+      });
   });
+});
 
-  describe("/api/users", () => {
-    test("GET: 200 Sends an array of user objects with the properties of username, name and avatar", () => {
-      return request(app)
-        .get("/api/users")
-        .expect(200)
-        .then(({ body }) => {
-          expect(body.length).toBe(4);
-          body.forEach((user) => {
-            expect(typeof user.username).toBe("string");
-            expect(typeof user.name).toBe("string");
-            expect(typeof user.avatar_url).toBe("string");
-          });
+describe("GET /api/articles", () => {
+  test("GET: 200 sends an array with articles", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles.length).toBe(13);
+        body.articles.forEach((article) => {
+          expect(typeof article.title).toBe("string");
+          expect(typeof article.topic).toBe("string");
+          expect(typeof article.author).toBe("string");
+          expect(typeof article.body).toBe("string");
+          expect(typeof article.created_at).toBe("string");
+          expect(typeof article.votes).toBe("number");
+        });
+      });
+  });
+  test("GET: 200 sends an array with articles filtered by topic", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles.length).toBe(12);
+        body.articles.forEach((article) => {
+          expect(article.topic).toBe("mitch");
         });
-    });
+      });
+  });
+  test("GET: 404 sends an appropriate error status alongside an error message when given a topic that does not exist in database", () => {
+    return request(app)
+      .get("/api/articles?topic=non-existent-topic")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Topic does not exist");
+      });
```

Comment:
There is one more test you can add! There is a topic in the database that exists but has no articles attached to it - this should return an empty array and 200 because it is valid. You already have the code set up to handle this but a test to prove it would be fab